### PR TITLE
feat(atomic): automatically adapt the number of element of the breadbox

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-924

### CODE REVIEW

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 1]
```diff
@@ -58,12 +62,67 @@ export class AtomicBreadbox implements InitializableComponent {
...
   @State() private isCollapsed = true;
 
-  // TODO: KIT-924 Automatically adapt the number of element of the breadbox
-  private collapseThreshold = 5;
-
   public initialize() {
     this.breadcrumbManager = buildBreadcrumbManager(this.bindings.engine);
     this.facetManager = buildFacetManager(this.bindings.engine);
+    this.resizeObserver = new ResizeObserver(() => this.adaptBreadcrumbs());
+    this.resizeObserver.observe(this.host.parentElement!);
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:69]: Inspired from the field values list, you'll see that I hide the elements with DOM manipulation (setting style display none & updating show more text) after:
- a resize of the list
- a re-render of the component

Another approach could have been to re-render multiple times instead, but it seamed to be less performant and more complex to manage.

Comment (ThibodeauJF) [reply packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:69]: That said, I'll explore a bit further using only JSX + internal state

Comment (samisayegh) [reply packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:69]: Would a combination of the two approaches be the simplest?
- Using `resizeObserver` to calculate which breadcrumbs to display,
- Setting this array in state e.g. `visibleBreadcrumbs`,
- Managing the render with JSX to display only the `visibleBreadcrumbs`, the correct show more count, aria-label

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx [hunk 2]
```diff
@@ -253,8 +323,22 @@ export class AtomicBreadbox implements InitializableComponent {
...
+            {this.renderBreadcrumbs()}
+          </ul>
+        </div>
       </div>
     );
   }
+
+  public componentDidRender() {
+    this.adaptBreadcrumbs();
+  }
```
Comment (samisayegh) [inline packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:343]: If the new changes were using JSX, then everything would be managed by the `render` call, removing the need for this life cycle method?

If there is a benefit of the current approach though, I would prefer if this life cycle method was closer to the top of the class (e.g. next to `disconnectedCallback`).

**Review by samisayegh — APPROVED** [review]
Not sure if you explored it, but I think leveraging JSX for the rendering could simplify the code in this case, with no noticeable performance difference

**Review by btaillon — APPROVED** [review]
Looks good!

File: packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx (hunk 1 above)
Comment (btaillon) [reply packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx:69]: If you end up using Sami's solution, please tag me and I'll create a JIRA to update the fields list component.

**Pull request merged**